File: Knjigoteka.Model/Requests/SaleInsert.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Knjigoteka.Model.Requests
{
    public class SaleInsert
    {
        public int EmployeeId { get; set; }
        public List<SaleItemInsert> Items { get; set; } = new();
    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/sale_insert.dart
class SaleInsert {
  final int employeeId;
  final List<SaleItemInsert> items;
  SaleInsert({required this.employeeId, required this.items});
  Map<String, dynamic> toJson() => {
    "EmployeeId": employeeId,
    "Items": items.map((i) => i.toJson()).toList(),
  };
}

class SaleItemInsert {
  final int bookId;
  final int quantity;
  SaleItemInsert({required this.bookId, required this.quantity});
  Map<String, dynamic> toJson() => {"BookId": bookId, "Quantity": quantity};
}
